server/app.mjs

```diff
@@ -369,35 +376,33 @@ const initGPTScriptConfig = async (gptscript) => {
             throw new Error(`Error parsing config file: ${parseErr}`);
         }
 
-        // Default values to add if they don't exist
-        const defaultConfig = {
-            gatewayURL: 'https://gateway-api.gptscript.ai',
-            integrations: {
-                microsoft365: 'microsoft365',
-                slack: 'slack',
-                notion: 'notion',
-                GitLab: 'gitlab'
-            }
-        };
-
-        // Update the config object with default values if they don't exist
-        config = {
-            ...defaultConfig,
-            ...config,
-            integrations: {
-                ...defaultConfig.integrations,
-                ...config.integrations
-            }
-        };
+        // Fetch default values to add if they don't exist
+        fetch("https://raw.githubusercontent.com/gptscript-ai/gateway-config/main/config.json")
+            .then((res) => res.json()
```

Comment:
IIUC, this is non-blocking, so the promise for initGPTScriptConfig will resolve before this promise has.

If I were to refactor this function I'd do something like:

```
const initGPTScriptConfig = async (gptscript) => {
        // Run a non-LLM no-op script to ensure the GPTScript config exists
        const run = await gptscript.evaluate({
            instructions: '#!sys.echo noop'
        })
        await run.text()
    
        const config = JSON.parse(await fs.readFile(configPath, 'utf8'))
        
        // Await the fetch request
        const response = await fetch('https://raw.githubusercontent.com/gptscript-ai/gateway-config/main/config.json');
        const defaultConfig = await response.json();

        // Assuming `config` is defined elsewhere in your module
        let config = {
            ...defaultConfig,
            ...config, // Merge with the existing config object
            integrations: {
                ...defaultConfig.integrations,
                ...config.integrations
            }
        };

        // Asynchronously write the updated config back to the file
        await fs.writeFile('config.json', JSON.stringify(config, null, 2), 'utf8');
```

Reply:
Fixed. We're using readFileSync now instead of readFile, since that fits what we want a lot better.